tuf/api/metadata.py

```diff
@@ -726,6 +726,10 @@ def __init__(
         super().__init__(version, spec_version, expires, unrecognized_fields)
         self.consistent_snapshot = consistent_snapshot
         self.keys = keys
+        mandatory_roles = ["root", "timestamp", "snapshot", "timestamp"]
+        if not (len(roles) == 4 and all(r in roles for r in mandatory_roles)):
```

Comment:
I admit I had to look up the docs for `all`. Wouldn't it be easier (and more performant) to do something like:
```suggestion
        if  set(roles) != TOP_LEVEL_ROLE_NAMES :
```

Note `TOP_LEVEL_ROLE_NAMES` could be a global `set` that contains all top-level role names and `roles` is a dictionary so we can rule out duplicates (don't have to check len).